### PR TITLE
Sync titan directory when adding blob files

### PR DESCRIPTION
**Pull request by yiwu-arbug**
Summary:
When adding new blob file, titan directory need to be fsync, otherwise files can be lost after power outage.

Test Plan:
run `strace -yfe fsync ./titan_titan_db_test --gtest_filter=*.Snapshot` and see titan directory being fsync-ed in the output.

Signed-off-by: Yi Wu <yiwu@pingcap.com>

### CODE REVIEW

**Comment by yiwu-arbug** [comment]
This one need to cherry-pick to 3.x branch.

File: src/blob_file_iterator_test.cc [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 #include "file/filename.h"
 #include "test_util/testharness.h"
+#include "util/random.h"
```
Comment (siddontang) [inline src/blob_file_iterator_test.cc]: why include this file?

Comment (yiwu-arbug) [reply src/blob_file_iterator_test.cc]: just to fix the build after updated rocksdb. too lazy to send a separate one liner fix..

Comment (yiwu-arbug) [reply src/blob_file_iterator_test.cc]: removed after rebase

**Pull request merged**